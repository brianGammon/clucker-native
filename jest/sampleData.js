const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
