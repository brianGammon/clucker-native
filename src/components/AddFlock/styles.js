import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  label: {
    marginBottom: 8,
    fontSize: 16,
  },
  input: {
    flex: 1,
    fontSize: 16,
    borderWidth: 1,
    borderColor: 'grey',
  },
});
