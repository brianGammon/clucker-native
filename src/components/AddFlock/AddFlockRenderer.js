import React from 'react';
import {
  Text, View, Button, TextInput,
} from 'react-native';
import styles from './styles';

type Props = {
  flockId: string,
  error: string,
  handleAddFlock: () => void,
  handleChangeText: (text: string) => void,
};

const AddFlockRenderer = ({
  flockId,
  error,
  handleAddFlock,
  handleChangeText,
}: Props) => (
  <View style={{ marginTop: 20 }}>
    <Text style={styles.label}>Add Flock</Text>
    <View style={styles.rowContainer}>
      <TextInput
        style={styles.input}
        value={flockId}
        onChangeText={text => handleChangeText(text)}
      />
      <Button title="Add" onPress={handleAddFlock} />
    </View>
    {error !== '' && <Text>{error}</Text>}
  </View>
);

export default AddFlockRenderer;
