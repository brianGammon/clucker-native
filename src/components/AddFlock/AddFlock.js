/* @flow */
import * as React from 'react';
import { connect } from 'react-redux';
import AddFlockRenderer from './AddFlockRenderer';
import { actionTypes } from '../../redux/constants';

type Props = {
  userId: string,
  addFlock: (userId: string, name: string) => void,
  error: string,
  inProgress: boolean,
};

type State = {
  name: string,
};

class AddFlock extends React.Component<Props, State> {
  state = { name: '' };

  componentDidUpdate(prevProps) {
    const { inProgress, error } = this.props;
    if (prevProps.inProgress && !inProgress && error === '') {
      this.resetForm();
    }
  }

  resetForm = () => {
    this.setState({ name: '' });
  };

  handleChangeText = (text: string) => {
    this.setState({ name: text });
  };

  handleAddFlock = () => {
    const { userId, addFlock } = this.props;
    const { name } = this.state;
    addFlock(userId, name);
  };

  render() {
    const { userId, error } = this.props;
    const { name } = this.state;
    return (
      <AddFlockRenderer
        userId={userId}
        handleAddFlock={this.handleAddFlock}
        handleChangeText={this.handleChangeText}
        name={name}
        error={error}
      />
    );
  }
}

const mapStateToProps = ({
  auth: { user },
  addForm: { error, inProgress },
}) => ({
  userId: user ? user.uid : '',
  error,
  inProgress,
});

const mapDispatchToProps = dispatch => ({
  addFlock: (userId: string, name: string) => dispatch({
    type: actionTypes.ADD_FLOCK_REQUESTED,
    payload: { userId, name },
  }),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AddFlock);
