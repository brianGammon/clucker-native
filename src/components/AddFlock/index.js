import AddFlock from './AddFlock';

export default AddFlock;
