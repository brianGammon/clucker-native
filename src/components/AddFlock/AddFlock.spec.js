import React from 'react';
import configureStore from 'redux-mock-store';
import sampleData from '../../../jest/test-data/full.2017-10-to-2018-05.json';
import AddFlock from '.';

describe('AddFlock component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(sampleData);
  test('Should render', () => {
    const wrapper = shallow(<AddFlock store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
