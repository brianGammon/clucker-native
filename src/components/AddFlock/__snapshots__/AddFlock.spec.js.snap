// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFlock component: Should render 1`] = `
<AddFlockRenderer
  error=""
  handleAddFlock={[Function]}
  handleChangeText={[Function]}
  name=""
  userId="user1"
/>
`;
